pkg/pushsync/pushsync.go

```diff
@@ -276,3 +255,36 @@ func (ps *PushSync) PushChunkToClosest(ctx context.Context, ch swarm.Chunk) (*Re
 
 	return rec, nil
 }
+
+func (ps *PushSync) deliverToPSS(ch swarm.Chunk) error{
+	// if callback is defined, call it for every new, valid chunk
+	errC := make(chan error)
+	defer close(errC)
+	if ps.deliveryCallback != nil {
+		go func () {
```

Comment:
What is the purpose of this goroutine and errC? With current code, it is just blocking or channel read and the equivalent code would be `return ps.deliveryCallback(ch)`.